gobblin-api/src/main/java/org/apache/gobblin/util/io/GsonInterfaceAdapter.java

```diff
@@ -189,7 +192,38 @@ private <S> S readValue(JsonObject jsonObject, TypeToken<S> defaultTypeToken) th
   }
 
   public static <T> Gson getGson(Class<T> clazz) {
-    Gson gson = new GsonBuilder().registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz)).create();
+    Gson gson = new GsonBuilder()
+        .setObjectToNumberStrategy(INTEGERNUMBERPOLICY.INTEGER_OR_LONG_OR_DOUBLE)
+        .registerTypeAdapterFactory(new GsonInterfaceAdapter(clazz))
+        .create();
     return gson;
   }
+
+  public enum INTEGERNUMBERPOLICY implements ToNumberStrategy {
```

Comment:
enums are named in camel case.. can rename enum to something like `CustomNumberStrategy / NumberHandlingPolicy`